modules/040-node-manager/hooks/update_node_group_status.go

```diff
@@ -417,6 +419,8 @@ func handleUpdateNGStatus(input *go_hook.HookInput) error {
 		)
 
 		patchNodeGroupStatus(input.PatchCollector, ngName, patch)
+		// set CR processed status
+		input.PatchCollector.Filter(set_cr_statuses.SetProcessedStatus, "deckhouse.io/v1", "nodegroup", "", ngName, object_patch.WithSubresource("/status"))
```

Comment:
So, deckhouse reacted and set the status, but what happens after helm? Does it set the same status to the same field?

My question is, how can I distinguish when the hook observes a new resource and when the helm chart, according to this new resource, was deployed?

Reply:
If my memory serves me well the problem that we couldn't resolve with LastObserved was that we can't use PatchCollector for applying LastObserved status if a hook fails to converge (if a user accidentally or deliberately provides inconsistent CR config). In other words:
* if a CR is valid than we have neat status fields (LastObserved and LastProcessed would match);
* if a CR is erroneous (for example, it was possible to create an alert manager CR, filling in non-existent kubernetes service as the destination) resulting in a failed hook, LastObserved status can't get updated because `PatchCollector` applies patches only after a successful hook run.
This limitation prevents us from setting an actual status for LastObserved in case there is an error. Moreover, it renders the CR as in sync because LastObserved and LastProcessed would match.

This is why we decided to give it a try with the generation field instead of LastObserved, even though it has nothing to do with deckhouse, it is still a kind of a marker that something was changed.